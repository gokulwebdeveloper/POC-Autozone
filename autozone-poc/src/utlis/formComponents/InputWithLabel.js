import React from "react";
import PropTypes from "prop-types";

export default function TextInput({
  labelText,
  inputId,
  inputType,
  isRequired,
  ...etc
}) {
  let inputLabel;
  switch (inputType) {
    case "text":
      inputLabel = (
        <>
          <label htmlFor={inputId}>
            {labelText} {isRequired ? <span>*</span> : null}
          </label>
          <input
            type="text"
            className="form-control"
            id={inputId}
            required={isRequired}
            defaultValue=""
            {...etc}
          />
        </>
      );
      break;

    case "email":
      inputLabel = (
        <>
          <label htmlFor={inputId}>
            {labelText} {isRequired ? <span>*</span> : null}
          </label>
          <input
            type="email"
            className="form-control"
            id={inputId}
            required={isRequired}
            defaultValue=""
            {...etc}
          />
        </>
      );
      break;

    case "checkbox":
      inputLabel = (
        <div className="custom-control custom-checkbox d-block mb-2">
          <input
            type="checkbox"
            className="custom-control-input"
            id={inputId}
            required={isRequired}
            {...etc}
          />
          <label className="custom-control-label" htmlFor={inputId}>
            {labelText} {isRequired ? <span>*</span> : null}
          </label>
        </div>
      );
      break;

    case "radio":
      inputLabel = (
        <div className="custom-control custom-radio d-block mb-2">
          <input
            type="radio"
            className="custom-control-input"
            id={inputId}
            required={isRequired}
            {...etc}
          />
          <label className="custom-control-label" htmlFor={inputId}>
            {labelText} {isRequired ? <span>*</span> : null}
          </label>
        </div>
      );
      break;
  }
  return inputLabel;
}

TextInput.propTypes = {
  labelText: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  inputId: PropTypes.string.isRequired,
  inputType: PropTypes.string.isRequired,
  isRequired: PropTypes.bool.isRequired,
};

TextInput.defaultProps = {
  labelText: "label",
  inputType: "text",
  isRequired: true,
};
