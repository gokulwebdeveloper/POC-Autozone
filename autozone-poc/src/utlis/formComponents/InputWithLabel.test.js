import React from 'react';
import renderer from 'react-test-renderer';
import {mount} from 'enzyme';
import InputWithLabel from './InputWithLabel';

describe("test InputWithLabel component", ()=>{
    //test if component render correctly
    it("component render correctly", ()=>{
        const input = mount(<InputWithLabel labelText="name" inputId="idtext" inputType="text" isRequired={true} />);
        expect(input.exists()).toBeTruthy();
        input.unmount();
    });

     //test if it should have 1 input and 1 label
     it("should have 1 input and 1 label", ()=>{
        const input = mount(<InputWithLabel labelText="name" inputId="idtext" inputType="text" isRequired={true} />);
        expect(input.find('input')).toHaveLength(1);
        expect(input.find('label')).toHaveLength(1);
        input.unmount();
    });

    //test if isRequired true span is added for asterik
    it("isRequired true then span is added for asterik", ()=>{
        const input = mount(<InputWithLabel labelText="name" inputId="idtext" inputType="text" isRequired={true} />);
        expect(input.find('span')).toHaveLength(1);
        input.unmount();
    });

    //test if label text set propely
    it("label text set propely", ()=>{
        const input = mount(<InputWithLabel labelText="name" inputId="idtext" inputType="text" isRequired={true} />);
        expect(input.find('label').text()).toBe('name *');
        input.unmount();
    });

    //test if props set propely in case of text input type
    it("props set propely in case of text input type", ()=>{
        const input = mount(<InputWithLabel labelText="name" inputId="idtext" inputType="text" isRequired={true} />);
        expect(input.prop('labelText')).toEqual('name');
        expect(input.prop('inputId')).toEqual('idtext');
        expect(input.prop('inputType')).toEqual('text');
        expect(input.prop('isRequired')).toBeTruthy();
        input.unmount();
    });

    //test if props set propely in case of email
    it("props set propely in case of email", ()=>{
        const input = mount(<InputWithLabel labelText="name" inputId="idtext" inputType="email" isRequired={false} />);
        expect(input.prop('labelText')).toEqual('name');
        expect(input.prop('inputId')).toEqual('idtext');
        expect(input.prop('inputType')).toEqual('email');
        expect(input.prop('isRequired')).toBe(false);
        input.unmount();
    });

    //test if props and additional props set propely in case of checkbox
    it("props set propely in case of email", ()=>{
        const input = mount(<InputWithLabel labelText="name" inputId="idtext" inputType="checkbox" isRequired={false} name="checkboxName" />);
        expect(input.prop('labelText')).toEqual('name');
        expect(input.prop('inputId')).toEqual('idtext');
        expect(input.prop('inputType')).toEqual('checkbox');
        expect(input.prop('isRequired')).toBe(false);
        
        //test for additional prop added
        expect(input.prop('name')).toBe('checkboxName');
        input.unmount();
    });

    //SnapShot Match test
    it("SnapShot Match", ()=>{
        const input = renderer.create(<InputWithLabel />).toJSON();
        expect(input).toMatchSnapshot()
    });
});