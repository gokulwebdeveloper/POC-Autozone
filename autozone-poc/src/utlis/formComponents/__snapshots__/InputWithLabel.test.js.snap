// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test InputWithLabel component SnapShot Match 1`] = `
Array [
  <label>
    label
     
    <span>
      *
    </span>
  </label>,
  <input
    className="form-control"
    defaultValue=""
    required={true}
    type="text"
  />,
]
`;
